src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = { ...state };
+  const resArr = [];
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(clone, obj.extraData);
+        resArr.push({ ...clone });
```

Comment:
you added this code 3 times. You can push updated clone object ones after switch case